receiver/k8seventsreceiver/k8s_event_to_logdata.go

```diff
@@ -0,0 +1,96 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8seventsreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8seventsreceiver"
+
+import (
+	"fmt"
+	"strings"
+
+	"go.opentelemetry.io/collector/model/pdata"
+	semconv "go.opentelemetry.io/collector/model/semconv/v1.5.0"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	// Number of log attributes to add to the pdata.LogSlice.
+	totalLogAttributes = 9
+
+	// Number of resource attributes to add to the pdata.ResourceLogs.
+	totalResourceAttributes = 3
+)
+
+// Only two types of events are created as of now.
+// For more info: https://docs.openshift.com/container-platform/4.9/rest_api/metadata_apis/event-core-v1.html
+var severityMap = map[string]pdata.SeverityNumber{
+	"normal":  pdata.SeverityNumberINFO,
+	"warning": pdata.SeverityNumberWARN,
+}
+
+// k8sEventToLogRecord converts Kubernetes event to pdata.LogSlice and adds the resource attributes.
+func k8sEventToLogData(logger *zap.Logger, ev *corev1.Event) pdata.Logs {
+	ld := pdata.NewLogs()
+	rl := ld.ResourceLogs().AppendEmpty()
+	ill := rl.InstrumentationLibraryLogs().AppendEmpty()
+	lr := ill.Logs().AppendEmpty()
+
+	resourceAttrs := rl.Resource().Attributes()
+	resourceAttrs.EnsureCapacity(totalResourceAttributes)
+
+	resourceAttrs.InsertString(semconv.AttributeK8SClusterName, ev.ObjectMeta.ClusterName)
+	resourceAttrs.InsertString("k8s.event.source", ev.Source.Component)
+	resourceAttrs.InsertString(semconv.AttributeK8SNodeName, ev.Source.Host)
+
+	// The Reason field is the most logical "name" of the event.
+	lr.SetName(ev.Reason)
```

Comment:
@hprajapati-splunk looks like the name field is going away https://github.com/open-telemetry/opentelemetry-specification/issues/2074

Let's use `k8s.event.reason` log attribute instead? Please add this attribute to your specification PR

Reply:
I have added the convention to the same PR and also used the same in this PR.